cube/src/main/java/zone/cogni/asquare/cube/hash/ModelHasher.java

```diff
@@ -0,0 +1,70 @@
+package zone.cogni.asquare.cube.hash;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Statement;
+import zone.cogni.asquare.cube.sort.SortedStatementsString;
+import zone.cogni.asquare.cube.sort.StringRdfVisitor;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.function.Function;
+
+public class ModelHasher implements Function<Model, byte[]> {
+
+    @Override
+    public byte[] apply(Model model) {
+        return hasBlankNode(model) ? hashFromSortedTurtle(model) : hashFromStatements(model);
+    }
+
+    private boolean hasBlankNode(Model model) {
+        return model.listStatements().toList().stream()
+                    .map(this::hasBlankNode)
+                    .reduce(Boolean::logicalOr)
+                    .orElse(false);
+    }
+
+    private boolean hasBlankNode(Statement stmt) {
+        return stmt.getSubject().isAnon() || stmt.getObject().isAnon();
+    }
+
+    private static byte[] hashFromSortedTurtle(Model model) {
+        SortedStatementsString sortedStatementsString = new SortedStatementsString();
+        String sortedString = sortedStatementsString.apply(model);
+
+        try {
+            MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+            byte[] bytes = sortedString.trim().getBytes(StandardCharsets.UTF_8);
```

Comment:
let's not trim... let's keep as is.

Reply:
I saw that the sortedString started with a newline, that's why I added it.

Reply:
let's fix the sorted string then :)